prototype/northbound/sensorthings/rest.api/src/main/java/org/eclipse/sensinact/sensorthings/sensing/rest/filters/OrderByFilter.java

```diff
@@ -43,17 +45,26 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
             throws IOException {
         @SuppressWarnings("unchecked")
         Comparator<Object> comparator = (Comparator<Object>) requestContext.getProperty(ORDERBY_PROP);
+        if (comparator == null) {
+            return;
+        }
 
         Object entity = responseContext.getEntity();
-        if(entity instanceof ResultList) {
-            ResultList<?> resultList = (ResultList<?>) entity;
-            resultList.value.sort(comparator);
+        if (entity instanceof ResultList) {
+            @SuppressWarnings("unchecked")
+            ResultList<Self> resultList = (ResultList<Self>) entity;
+            List<Self> sortedList = new ArrayList<>(resultList.value);
+            sortedList.sort(comparator);
+            resultList.value = List.copyOf(sortedList);
         }
     }
 
     @Override
     public void filter(ContainerRequestContext requestContext) throws IOException {
         List<String> list = requestContext.getUriInfo().getQueryParameters().get("$orderby");
+        if (list == null) {
```

Comment:
This behaviour is wrong. In the absence of a sort property we should default to sorting by id as per the spec